### PR TITLE
Use new IDFv4 Cmake `idf_component_register` API (VSC-240)

### LINKED ISSUE #31: Need to update app-template (VSC-236)
**Describe the bug**
CMakefile.txt is using old components registration API, which doesn't have document support.

**To Reproduce**

Press F1 and type ESP-IDF: Create ESP-IDF project to generate a template ESP-IDF project.

**Expected behavior**
I suppose CMakefile.txt should be using `idf_component_register `instead of `register_component`

**Screenshots**
image

**Environment (please complete the following information):**
OS Name & Version: [Win10]
VSCode Version: [1.42]
ESP-IDF Version: [4.1]
Python Version: [3.7]

**Additional context**
https://github.com/espressif/esp-idf-template

### PR DESCRIPTION
**Pull request by SoumeshBanerjee**
Closes #31

### CODE REVIEW

File: templates/template-app/main/CMakeLists.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ set(COMPONENT_PRIV_REQUIRES )
 set(COMPONENT_SRCS "main.c")
 set(COMPONENT_ADD_INCLUDEDIRS "")
 
-register_component()
\ No newline at end of file
+idf_component_register()
```
Comment (igrr) [inline templates/template-app/main/CMakeLists.txt]: The list of source and include directories should be passed to idf_component_register. You can refer to https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/build-system.html#minimal-component-cmakelists

File: templates/template-app/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+build/
```
Comment (igrr) [inline templates/template-app/.gitignore]: Recommend adding `sdkconfig` and `sdkconfig.old` here.

File: templates/template-app/main/CMakeLists.txt (hunk 1 above)
Comment (SoumeshBanerjee) [reply templates/template-app/main/CMakeLists.txt]: Added in #38

File: templates/template-app/.gitignore (hunk 2 above)
Comment (SoumeshBanerjee) [reply templates/template-app/.gitignore]: Added in #38

**Pull request merged**